main/src/main/scala/sbt/Script.scala

```diff
@@ -32,7 +42,7 @@ object Script {
       val append = Load.transformSettings(Load.projectScope(currentRef), currentRef.build, rootProject, scriptSettings ++ embeddedSettings)
 
       val newStructure = Load.reapply(session.original ++ append, structure)
-      val arguments = state.remainingCommands.drop(1)
+      val arguments = state.remainingCommands.drop(1).map(arg => "\"" + arg + "\"")
```

Comment:
I checked `run` and quoting the argument works from the sbt prompt.
```
> run "sbt World"
[info] Running Test sbt World
[info] Hello, sbt World!
```
Running from sbt requires escaping the quotes:
```
sbt "run \"sbt World\""
[info] Running Test sbt World
[info] Hello, sbt World!
```